webserver.sh

```diff
@@ -44,28 +44,30 @@ else
   ssh-keygen -t ed25519 -f "./id_ed25519" -N ''
 fi
 
-# checking for cloud-config.yaml
-if [ -f "./cloud-config.yaml" ]
+# checking for cloud-init.yaml
+if [ -f "./cloud-init.yaml" ]
 then
-  echo "cloud-config.yaml file already exists"
-else
-  echo "cloud-config.yaml file does not exist ... creating"
-  cat <<- _EOF_ > ./cloud-config.yaml
-	# cloud-config
-	users:
-	  - name: $USER
-	    ssh-authorized-keys:
-	      - $(cat ./id_ed25519.pub)
-	_EOF_
+  echo "cloud-init.yaml file already exists...deleting"
+  rm ./cloud-init.yaml
```

Comment:
Why would you delete the file if you need it to be there and then create it outside of the conditional?

Reply:
We discussed during class that just because a file exists, does not ensure the content/state of the file is what’s desired.

An alternative approach would be to test the contents, or simply overwrite the contents at script runtime versus always deleting + recreating.

Reply:
Right. I was hoping we'd just overwrite if it's not there but then as a next level thing, test the contents based on the existing pub key in the conditional.

Reply:
This would be a good improvement!